solidity/contracts/middleware/libs/InterchainAccountMessage.sol

```diff
@@ -34,13 +41,15 @@ library InterchainAccountMessage {
     ) internal pure returns (bytes memory) {
         CallLib.Call[] memory _calls = new CallLib.Call[](1);
         _calls[0] = CallLib.build(_to, _value, _data);
-        return
-            abi.encode(
-                TypeCasts.addressToBytes32(_owner),
-                _ism,
-                _calls,
-                EMPTY_SALT // Salts are expected when decoding.
-            );
+
+        bytes memory prefix = abi.encodePacked(
+            MessageType.CALLS,
+            _owner.addressToBytes32(),
+            _ism,
+            EMPTY_SALT
+        );
+        bytes memory suffix = abi.encode(_calls);
+        return bytes.concat(prefix, suffix);
```

Comment:
why use `bytes.concat` instead of `abi.encodePacked`? im actually not sure what the difference is

Reply:
See discussion here: https://github.com/ethereum/solidity/issues/10903